src/WebJobs.Script/Extensions/IServiceProviderExtensions.cs

```diff
@@ -21,11 +21,19 @@ public static class IServiceProviderExtensions
                 throw new ArgumentNullException(nameof(serviceProvider));
             }
 
-            var hostManager = serviceProvider.GetService<IScriptHostManager>();
-            if (Utility.TryGetHostService(hostManager, out TService service))
+            try
             {
-                return service;
+                var hostManager = serviceProvider.GetService<IScriptHostManager>();
+                if (Utility.TryGetHostService(hostManager, out TService service))
+                {
+                    return service;
+                }
             }
+            catch
```

Comment:
do we need to catch only "disposing' exceptions

Reply:
I had done that originally, but decided to use the same handling that we do elsewhere. Also, another issue with trying to handle specific types is the DryIoc.ContainerException is defined in WebHost which isn't accessible here, so I'd have to catch the base InvalidOperation type. Rather than do that I think what I have here is better.